elasticdl/python/common/model_utils.py

```diff
@@ -179,3 +181,30 @@ def get_non_embedding_trainable_vars(model, embedding_layers):
         if not is_embedding_item:
             non_embedding_trainable_vars.append(var)
     return non_embedding_trainable_vars
+
+
+def get_optimizer_info(optimizer):
+    """
+    Get optimizer type and its argument values
+    """
+    # TODO: add more supported optimizers
+    OPT_ARGUMENTS = {
```

Comment:
This is very hard to maintain though. Also if TF introduces new optimizers, we have to add them here. Backwards compatibility is also hard to guarantee.

Reply:
This is just an interface for TF. We will maintain this interface with the tf version we supported in Dockerfile. 
All optimizers are standard computations. For example, TF SGD optimizer contains 2 optimizers,  standard SGD and Nesterov variation, depending on `nesterov` argument. 
TF Adam optimizer also contains 2 optimizers, standard Adam and its `amsgrad` variation.

In Golang PS, we will implement these 4  optimizers, and select one depending on the input arguments. In this way, our Golang PS can support other ML frameworks such as PyTorch, as long as the optimizer type and arguments are given.

For backwards compatibility, for the interface, we only support the TF version used in Dockerfile. For optimizer computation, these are standard algorithms used by all ML frameworks.